config/common/common.go

```diff
@@ -93,3 +103,102 @@ func GetFullyQualifiedIDFn(serviceProvider string, keyPairs ...string) tjconfig.
 		return path, nil
 	}
 }
+
+var referenceRules = [][]string{
+	{"resource_group$", "/azure/v1alpha1.ResourceGroup"},
+	{"subnet$", "/network/v1alpha1.Subnet"},
+}
+
+// AddCommonReferences adds some common reference fields.
+// This is a part of resource generation pipeline.
+func AddCommonReferences(references tjconfig.References, resource *schema.Resource, shortGroup string, nestedFieldNames []string) error {
```

Comment:
nit: Because this function accepts 3 fields from a `*config.Resource`, I think we can change the signature so that it accepts a `*config.Resource` for the sake of conciseness:
```suggestion
func AddCommonReferences(r *config.Resource, nestedFieldNames []string) error {
```

Reply:
We need to pass the `*schema.Resource` object (that is part of the `*tjconfig.Resource`) to recursive call explicitly. Because we processed the root level schema.Resource in first call and then in recursive call we want to check the second level TerraformedResource. However, tjconfig.Resource still has the root level resource.

I changed the signature of exported function to: `AddCommonReferences(r *config.Resource)`
However, I protected the current signature in unexported function: `addCommonReferences(references tjconfig.References, resource *schema.Resource, shortGroup string, nestedFieldNames []string)`